### DISCUSSION TITLE
IINA as default player

### DISCUSSION THREAD

**Discussion by donnydark0** [streamlink/streamlink#4498] [Q&A] [answered]
Hi,

I am having issues changing the default player from VLC to IINA. I managed to do so in Twitch GUI, but I would like to open Twitch Streams in Chatterino with IINA as well.

I already tried creating a config file with some lines of settings I found on the web, but could not get it to work.

Does anyone know how to change the default player on MacOS?

**Comment by bastimeyer** [comment, marked as answer]
> with some lines of settings I found on the web, but could not get it to work.

**Post your entire config file.**
"some lines" and "could not get it to work" is useless information.

In the IINA feature request thread on the Streamlink Twitch GUI issue tracker (you're probably referring to), you can see that IINA doesn't support reading from `stdin` unless its `--stdin` parameter is set (which is pretty bad):
https://github.com/streamlink/streamlink-twitch-gui/issues/480#issuecomment-882122375

Streamlink always will add the dash character `-` to the player's arguments if the stream data is fed to the player via standard I/O streams, which is UNIX style for telling applications to read from `stdin`. A workaround for this is setting Streamlink's `--player-args` to `{playerinput}-stdin`, which results in `--stdin` being set as player args instead of `-`. See Streamlink's `--player-args` documentation:
- https://streamlink.github.io/latest/cli.html#cmdoption-player-args

Otherwise, try using other player options, like `--player-continuous-http` or `--player-external-http`, so that the player doesn't have to read from `stdin` by from a locally spawned HTTP server instead:
- https://streamlink.github.io/latest/cli.html#cmdoption-player-continuous-http
- https://streamlink.github.io/latest/cli.html#cmdoption-player-external-http

And in case that IINA is capable for playing back HLS streams directly, you can set `--player-passthrough=hls`, but this will bypass Streamlink completely:
- https://streamlink.github.io/latest/cli.html#cmdoption-player-passthrough

**Reply by donnydark0** [reply]
Here's the solution since this question in highly ranked in google for "chatterino iina":

streamlink --player=/Applications/IINA.app/Contents/MacOS/iina-cli --player-args {playerinput}-stdin https://twitch.tv/example best

To get custom path to Streamlink enter "which streamlink" in Terminal.

Chatterino:
<img width="800" alt="Screenshot 2022-07-18 at 17 03 39" src="https://user-images.githubusercontent.com/53871425/179541737-84564167-aefc-4d14-ac9b-e06be532834d.png">
